### PR TITLE
[fuchsia] Bump the target API level to 12, and pass it to fidlc

### PR DESCRIPTION
**Pull request by mk12**
This makes the GN build pass the target Fuchsia API level to fidlc. Currently it relies on the default of HEAD, meaning all FIDL changes are immediately visible when the SDK rolls. This was never the intention: flutter should be targeting a specific API level.

This unblocks https://fxrev.dev/864297 which makes the fidlc --available flag required, and which uncovered this problem.

I also copied the changes to gen_response_file.py from https://fxrev.dev/865020 to make it forward --available to fidlc.

### CODE REVIEW

File: tools/fuchsia/fidl/fidl_library.gni [hunk 1]
```diff
@@ -22,6 +22,9 @@ template("fidl_library") {
 
   assert(defined(invoker.sources), "A FIDL library requires some sources.")
 
+  assert(fuchsia_target_api_level != -1,
```
Comment (tarobins) [inline tools/fuchsia/fidl/fidl_library.gni:25]: are there existing usages of this template?  is fuchsia_target_api_level set for all on them?

**Comment by arbreng** [comment]
Generally LGTM, but before merging I pinged mitchell offline with some questions I have

File: tools/fuchsia/fidl/fidl_library.gni (hunk 1 above)
Comment (mk12) [reply tools/fuchsia/fidl/fidl_library.gni:25]: Yes it's used to make FIDL bindings for Fuchsia SDK libraries (see tools/fuchsia/sdk/sdk_targets.gni). I copied the != -1 assertion from tools/fuchsia/fuchsia_archive.gni. `fuchsia_target_api_level` is a global GN arg set here https://github.com/flutter/engine/blob/main/tools/gn#L570. The build should fail if there's a case where it's not set.

**Comment by mk12** [comment]
Amended to also bump the API level to 12 as discussed with @arbreng

**Comment by tarobins** [comment]
LGTM

**Comment by mk12** [comment]
@tarobins It says you "self-requested a review" - did you forget to approve, or are you waiting for @arbreng?

**Review by tarobins — APPROVED** [review]

**Comment by tarobins** [comment]
No, I just don't know what I'm doing.

**Comment by mk12** [comment]
👍 Thanks. Looks like I still can't merge even with approval, could you do that as well?

**Review by arbreng — APPROVED** [review]

**Pull request merged**